### PR TITLE
Configurable tx limit

### PR DESCRIPTION
**Pull request by oleks-rip**

### CODE REVIEW

File: src/xbwd/client/ChainListener.cpp [hunk 1]
```diff
@@ -54,15 +54,19 @@ ChainListener::ChainListener(
     std::optional<ripple::AccountID> submitAccount,
     std::weak_ptr<Federator>&& federator,
     std::optional<ripple::AccountID> signAccount,
+    std::uint32_t txLimit,
+    unsigned lastLedgerProcessed,
```
Comment (seelabs) [inline src/xbwd/client/ChainListener.cpp:58]: This is a nit, but I prefer `std::uint` types over `unsigned`. I know the history processor is already using `unsigned` so it makes sense to do it here to, but we should make a note to clean that up.

File: src/xbwd/client/ChainListener.cpp [hunk 2]
```diff
@@ -1393,6 +1407,18 @@ ChainListener::getSubmitProcessedLedger() const
     return ledgerProcessedSubmit_;
 }
 
+std::atomic_uint32_t const&
```
Comment (seelabs) [inline src/xbwd/client/ChainListener.cpp:1410]: I see we're return an atomic. Is there a reason we can't return a `std::uint32_t` and do the `load` in this function? It looks like all the callers are going to immediately `load` the value anyway.

File: src/xbwd/client/ChainListener.h [hunk 3]
```diff
@@ -175,6 +184,12 @@ class ChainListener : public std::enable_shared_from_this<ChainListener>
     std::uint32_t
     getHistoryProcessedLedger() const;
 
+    std::atomic_uint32_t const &
```
Comment (seelabs) [inline src/xbwd/client/ChainListener.h:187]: Formatting

File: src/xbwd/client/ChainListener.h [hunk 4]
```diff
@@ -175,6 +184,12 @@ class ChainListener : public std::enable_shared_from_this<ChainListener>
...
+    getCurrentLedger() const;
+
+    std::atomic_uint32_t const &
```
Comment (seelabs) [inline src/xbwd/client/ChainListener.h:190]: Formatting

File: src/xbwd/federator/Federator.cpp [hunk 5]
```diff
@@ -1333,17 +1333,17 @@ Federator::onEvent(event::NewLedger const& e)
...
     {
+        dbTTL = false;
         std::lock_guard l{txnsMutex_};
         auto& subs = submitted_[ct];
         for (auto& s : subs)
             if (!s->lastLedgerSeq_)
-                s->lastLedgerSeq_ = e.ledgerIndex_ + TxnTTLLedgers;
+                s->lastLedgerSeq_ =
+                    chains_[ct].listener_->getCurrentLedger() + TxnTTLLedgers;
     }
```
Comment (seelabs) [inline src/xbwd/federator/Federator.cpp:1347]: Maybe something like this instead?
```
    static bool _ = [&] {
        std::lock_guard l{txnsMutex_};
        auto& subs = submitted_[ct];
        for (auto& s : subs)
            if (!s->lastLedgerSeq_)
                s->lastLedgerSeq_ =
                    chains_[ct].listener_->getCurrentLedger() + TxnTTLLedgers;
        return true;  // not used
    }();

```

File: src/xbwd/federator/Federator.cpp [hunk 6]
```diff
@@ -2005,7 +2016,7 @@ Json::Value
 Federator::getInfo() const
 {
     // TODO
-    // Track transactons per secons
+    // Track transactions per secons
```
[the remark this answers is not in the document]
Comment (oleks-rip) [reply src/xbwd/federator/Federator.cpp]: fixed a377ea06aa312b3864e

File: src/xbwd/client/ChainListener.cpp (hunk 2 above)
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.cpp]: fixed a377ea06aa312b3864e

File: src/xbwd/client/ChainListener.h (hunk 3 above)
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.h]: fixed a377ea06aa312b3864e

File: src/xbwd/client/ChainListener.h (hunk 4 above)
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.h]: fixed a377ea06aa312b3864e

File: src/xbwd/federator/Federator.cpp (hunk 5 above)
Comment (oleks-rip) [reply src/xbwd/federator/Federator.cpp]: fixed a377ea06aa312b3864e

**Review by seelabs — APPROVED** [review]

**Pull request merged**